internal/datacoord/meta.go

```diff
@@ -1977,35 +1980,41 @@ func (m *meta) SaveStatsResultSegment(oldSegmentID int64, result *workerpb.Stats
 	updateSegStateAndPrepareMetrics(cloned, commonpb.SegmentState_Dropped, metricMutation)
 
 	segmentInfo := &datapb.SegmentInfo{
```

Comment:
How about using proto.Clone and then modifying individual members?

Reply:
There is already a cloned segment because the old segment also needs to be modified.